src/@shared/Modal.tsx

```diff
@@ -0,0 +1,21 @@
+import { useModal, useModalState } from "./useModalStore";
+import * as css from "./Modal.css";
+
+export function Modal() {
+  const { isOpen, ModalComponent, location, animation } = useModalState();
+  const { offModal } = useModal();
+
+  return isOpen ? (
```

Comment:
```suggestion
  return isOpen && (
```

&& 을 쓰면 null 대신 undefined가 return 되는 차이가 있긴 한데, 삼항연산자는 어떤 두 가지의 분기가 있는 경우를 처리하면 좋을 것 같아요. 지금같은 경우는 특정 조건에서 렌더링 되는 경우니까 && 연산자가 조금 더 적합하지 않을까 하네요..!

Reply:
이 부분은 modal을 사용하는 컴포넌트 내에서 Modal 컴포넌트를 사용할 때 아무것도 return하지 않으면 JSX에 맞지 않는다는 에러가 나서 일부로 null을 추가해줬어요. Modal 컴포넌트 넣을 때 조건부 처리해주고 싶지 않아서요~